packages/itwinui-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -412,33 +408,43 @@ export const ComboBox = React.forwardRef(
           return;
         }
 
-        if (isMultipleEnabled(selected, multiple)) {
+        if (isMultipleEnabled(value, multiple)) {
           const actionType = isMenuItemSelected(__originalIndex)
             ? 'removed'
             : 'added';
-          const newArray = selectedChangeHandler(__originalIndex, actionType);
-          dispatch({ type: 'multiselect', value: newArray });
+          const newArray = selectedChangeHandler(
+            __originalIndex,
+            actionType,
+            value,
+          );
+          setValue(newArray);
           onChangeHandler(__originalIndex, actionType, newArray);
 
           // update live region
+          // TODO: Confirm if need to show live region when controlled state and the onChangeProp does not update the value
           setLiveRegionSelection(
             newArray
-              .map((item) => optionsRef.current[item]?.label)
+              .map(
+                (item) =>
+                  optionsRef.current.find((option) => option.value === item)
+                    ?.label,
+              )
```

Comment:
I'm wondering when we need to set the live region. Do we need set it when the `value` changes externally (so that the user is aware that the value has changed) or whenever the user explicitly selects/deselects an option?

If it's the latter, how to implement this in the controlled state? Example case: the user selects/deselects an item but the application does not update the state in the `onChange`.

In this example case, should we set the live region? If not, then how do we know when the application has updated the `value` after the user has selected/deselected the item vs. when the `value` has changed externally without the user selecting/deselecting an option?

Reply:
The live region should update whenever the value actually changes.

The point of controlled state is not to allow applications to ignore the user's actions, but to instead have an _additional_ way to set the value. If the application does not respect the state provided by `onChange`, that's a horrendous bug (or a great way to annoy users).

Reply:
Sounds good. So, I will leave the `setLiveRegionSelection` where it is currently in the `onClickHandler` since users should always immediately update the controlled value whenever the `onChangeHandler` is called.